BibBuddy/app/src/main/java/de/bibbuddy/SearchFragment.java

```diff
@@ -1,20 +1,326 @@
+    context = view.getContext();
+
+    ((MainActivity) getActivity()).updateHeaderFragment(getString(R.string.navigation_search));
+
+    requireActivity().getOnBackPressedDispatcher().addCallback(new OnBackPressedCallback(true) {
+      @Override
+      public void handleOnBackPressed() {
+
+        FragmentManager fm = getParentFragmentManager();
+        if (fm.getBackStackEntryCount() > 0) {
+          fm.popBackStack();
+        } else {
+          requireActivity().onBackPressed();
+        }
+      }
+    });
+
+    setupSearchInput();
+    setupRecyclerView();
+    setupSearchButton();
+
+    setHasOptionsMenu(true);
+    sortCriteria = SearchSortCriteria.MOD_DATE_LATEST;
+
     return view;
   }
+
+  @Override
+  public void onCreateOptionsMenu(@NonNull Menu menu, MenuInflater inflater) {
+    inflater.inflate(R.menu.fragment_search_menu, menu);
+    super.onCreateOptionsMenu(menu, inflater);
+  }
+
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    switch (item.getItemId()) {
+      case R.id.menu_search_sort:
+        handleSearchSort();
+        break;
+
+      case R.id.menu_search_filter:
+        handleSearchFilter();
+        break;
+
+      case R.id.menu_search_help:
+        handleHelp();
+        break;
+
+      default:
+    }
+
+    return super.onOptionsItemSelected(item);
+  }
+
+
+  private void setupRecyclerView() {
+    searchRecyclerView = view.findViewById(R.id.search_recycler_view);
+
+    searchModel = new SearchModel(context);
+    searchResultList = new ArrayList<>();
+
+    // TODO String is always empty when this fragment is newly created even if the text is filled in
+    if (!DataValidation.isStringEmpty(searchInput.getText().toString())) {
+      searchItems();
+    }
+
+    adapter = new SearchRecyclerViewAdapter(searchResultList, this);
+    searchRecyclerView.setAdapter(adapter);
+
+    updateEmptyView(searchResultList);
+  }
+
+  private void updateEmptyView(List<SearchItem> searchResultList) {
+    TextView emptyView = view.findViewById(R.id.list_view_search_empty);
+
+    if (searchResultList.isEmpty()) {
+      emptyView.setVisibility(View.VISIBLE);
+    } else {
+      emptyView.setVisibility(View.GONE);
+    }
+  }
+
+  private void setupSearchButton() {
+    ImageButton searchBtn = view.findViewById(R.id.search_btn);
+
+    searchBtn.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View v) {
+        searchItems();
+      }
+    });
+  }
+
+  private void setupSearchInput() {
+    searchInput = view.findViewById(R.id.search_input);
+
+    searchInput.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+      @Override
+      public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+        searchItems();
+        return false;
+      }
+    });
+  }
+
+  private void searchItems() {
+    String searchInputStr = searchInput.getText().toString();
```

Comment:
```suggestion
    String searchInput = searchInput.getText().toString();
```

Reply:
I can't use searchInput there because this variable is already used